src/App.tsx

```diff
@@ -1,14 +1,42 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [people] = useState(peopleFromServer);
+  const [hasClick, setHasClick] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(
+      person => person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [people, appliedQuery]);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
try to implement this requirement:
![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/b3cf1749-0ed8-46e6-80d7-0d65413e3997)

hint: you have to somehow change with a delay one more state variable (like hasClick)